near-sdk/src/store/lookup_set/mod.rs

```diff
@@ -92,7 +167,38 @@ where
     where
         T: Clone,
     {
-        self.map.insert(value, ()).is_none()
+        let entry = self.get_mut_inner(&value);
+        match entry {
+            EntryState::Inserted | EntryState::Present => false,
+            EntryState::Deleted => {
+                let _ = std::mem::replace(entry, EntryState::Present);
+                true
+            }
+            EntryState::Absent => {
+                let _ = std::mem::replace(entry, EntryState::Inserted);
```

Comment:
```suggestion
                *entry = EntryState::Present;
                true
            }
            EntryState::Absent => {
                *entry = EntryState::Inserted;
```

Don't need `mem::replace` if you don't want to use the existing value (you are ignoring here)

Reply:
Indeed, somehow I forgot that you could do that :)